frontend/src/utils/umbra-api.ts

```diff
@@ -10,17 +10,42 @@ import {
   RelayResponse,
   TokenListResponse,
   WithdrawalInputs,
+  UmbraApiVersion,
 } from 'components/models';
 import { jsonFetch } from 'src/utils/utils';
 
+import useSettingsStore from 'src/store/settings';
+
+const { getUmbraApiVersion, setUmbraApiVersion, clearUmbraApiVersion } = useSettingsStore();
+
 export class UmbraApi {
   static baseUrl = 'https://mainnet.api.umbra.cash'; // works for all networks
+  // static baseUrl = 'http://localhost:3000';  // use this for testing with a local Umbra API
   constructor(
     readonly tokens: TokenInfoExtended[],
     readonly chainId: number,
     readonly nativeTokenMinSendAmount: string | undefined
   ) {}
 
+  static checkUmbraApiVersion(version: UmbraApiVersion) {
+    const apiVersionFromSettings: UmbraApiVersion | null = getUmbraApiVersion();
```

Comment:
I think the return type can be removed here. It is better to be less explicit with types in typescript as these can hide type errors.

Reply:
Oh interesting, is that a general tip for typescript? I thought using implicit types can hide type errors.

Reply:
Yes, in my experience explicit types can lead to mistyping

Reply:
Done: https://github.com/ScopeLift/umbra-protocol/pull/653/commits/c64845cd8120990d046476de35b3791f89ac1271